disp-reconstruct.py

```diff
@@ -0,0 +1,249 @@
+        mask = (~np.isnan(hillas['width'])
+                * ~np.isnan(hillas['cen_x']) * mask0 * mask1 * mask2)
+    else:
+        mask = (~np.isnan(hillas['width'])
+                * ~np.isnan(hillas['cen_x']) * mask1 * mask2)
+
+    # hillas
+    size = hillas['size'][mask]
+    width = hillas['width'][mask]
+    length = hillas['length'][mask]
+    psi = hillas['psi'][mask]
+    cog_x = hillas['cen_x'][mask]  # in mm
+    cog_y = hillas['cen_y'][mask]  #
+    skewness = hillas['skewness'][mask]
+
+    # image
+    # there is event number in the first column, the rest
+    # are dl1_camera.pe_samples values that survived cleaning
+    image = image[mask, 1:]
+
+    x_offset = 0*np.ones(len(cog_x))    # MC event source position in deg
+    y_offset = -float(data_offset)*np.ones(len(cog_x))
+
+    print('N of events before cuts:, ', len(hillas['size']))
+    print('N of events after cuts: ', len(hillas['size'][mask]))
+
+    # conversion of coordinates in mm to deg.
+    # Digicam: 0.24 deg/px, one SiPM is 24.3 mm wide
+    mm_to_deg = 0.24 / 24.3
+    cog_x = cog_x * mm_to_deg   # conversion to degrees
+    cog_y = cog_y * mm_to_deg   # conversion to degrees
+
+    # Outermost pixels
+    (leakage2, pix_bound, image_mask,
+     signal_full, signal_border) = leak_pixels(pix_x, pix_y, image)
+
+    # Interpolation in lookup table (need to be done for each
+    # azimuth separately or regular grid interpolation have to be implemented)
+    A = Parameters()
+    if equation == 1:
+        f = interp2d(lookup['zenith'],
+                     lookup['offset'], lookup['A0'], kind='linear')
+        a0 = f(data_zenith, data_offset)
+        A.add('A0', value=a0)
+
+    elif equation == 4:
+        f1 = interp2d(lookup['zenith'],
+                      lookup['offset'], lookup['A0'], kind='linear')
+        f2 = interp2d(lookup['zenith'],
+                      lookup['offset'], lookup['A1'], kind='linear')
+        a0 = f1(data_zenith, data_offset)
+        a1 = f2(data_zenith, data_offset)
+        A.add_many(('A0', a0), ('A1', a1))
+
+    elif equation == 5:
+        f1 = interp2d(lookup['zenith'],
+                      lookup['offset'], lookup['A0'], kind='linear')
+        f2 = interp2d(lookup['zenith'],
+                      lookup['offset'], lookup['A1'], kind='linear')
+        f3 = interp2d(lookup['zenith'],
+                      lookup['offset'], lookup['A2'], kind='linear')
+        a0 = f1(data_zenith, data_offset)
+        a1 = f2(data_zenith, data_offset)
+        a2 = f3(data_zenith, data_offset)
+        A.add_many(('A0', a0), ('A1', a1), ('A2', a2))
+
+    print(A)
+    (disp_comp, x_source_comp,
+     y_source_comp, residuals) = disp_eval(A, width, length, cog_x, cog_y,
+                                           x_offset, y_offset, psi, skewness,
+                                           size, leakage2, equation)
+
+    # ARRIVAL DIRECTION DISTRIBUTION
+    bins = 100
+
+    if args['--modification']:
```

Comment:
`args` is not defined in this function. This is not a syntax error, because `args` is a global variable defined in the `if name == main - block` .... however using global vars is not very good style .. (I do it myself sometimes ... but I think you can simply use `modification` here, no?

Reply:
Right :), corrected.